client/core/notification.go

```diff
@@ -56,6 +63,26 @@ func (c *Core) notify(n Notification) {
 	c.noteMtx.RUnlock()
 }
 
+// updateCoreMatch performs a targeted update of a match in the User struct.
+func (c *Core) updateCoreMatch(n *MatchNote) {
```

Comment:
This function and the way it's called from the notification feed might be a model for a solution to https://github.com/decred/dcrdex/issues/379.